### ISSUE TITLE
Consider adding a message in the log when a package got published

### ISSUE THREAD

**Issue by hasezoey** [semantic-release-cargo/semantic-release-cargo#372]
Currently this package seemingly logs nothing on `publish` if successful, maybe consider adding a message along the lines of `Published version to crates.io: https://crates.io/crates/CRATE/X.X.X`, similar to how other plugins do it (including github, npm, etc).

current text:
```txt
[6:01:14 PM] [semantic-release] › ℹ  Start step "publish" of plugin "@semantic-release-cargo/semantic-release-cargo"
[6:01:21 PM] [semantic-release] › ✔  Completed step "publish" of plugin "@semantic-release-cargo/semantic-release-cargo"
```

example from plugin github:
```txt
[6:01:21 PM] [semantic-release] › ℹ  Start step "publish" of plugin "@semantic-release/github"
[6:01:21 PM] [semantic-release] [@semantic-release/github] › ℹ  Published GitHub release: https://github.com/hasezoey/new_string_template/releases/tag/v1.5.0
[6:01:21 PM] [semantic-release] › ✔  Completed step "publish" of plugin "@semantic-release/github"
```

example from plugin npm:
```txt
[12:37:08 PM] [semantic-release] › ℹ  Start step "publish" of plugin "@semantic-release/npm"
[12:37:08 PM] [semantic-release] [@semantic-release/npm] › ℹ  Publishing version 12.3.0 to npm registry on dist-tag latest
npm notice 
... a bunch of npm notice
+ @typegoose/typegoose@12.3.0
[12:37:12 PM] [semantic-release] [@semantic-release/npm] › ℹ  Published @typegoose/typegoose@12.3.0 to dist-tag @latest on https://registry.npmjs.org/
[12:37:12 PM] [semantic-release] › ✔  Completed step "publish" of plugin "@semantic-release/npm"
```

---

i am not 100% sure as to what version of the plugin is used, but i think it is `2.2.96` (i dont know as it is not listed in the logs and i am using `cycjimmy/semantic-release-action`)

**Comment by ncatelli** [comment]
Yeah I'd second this one. It'd be nice if it included atleast, but not limited 2, an info level log that functions a bit like the `npm` plugin example. Currently the [debug](https://github.com/semantic-release-cargo/semantic-release-cargo/blob/fb57849f074195b5c1861fcca507018f9bad0748/src/lib.rs#L649) macros are functioning as the only logging, and arguably closer to `info` level logging in many places.

I'd suggest that we follow the npm format, while defaulting to info log levels. Where messages follow a few of the following examples

### Info Logs
#### Prepare stage
Similarly to the [debug logs for the prepare steps](https://github.com/semantic-release-cargo/semantic-release-cargo/blob/fb57849f074195b5c1861fcca507018f9bad0748/src/lib.rs#L314) these would probably be better served as info logs.

```
[12:37:08 PM] [semantic-release] [@semantic-release-cargo/semantic-release-cargo] › ℹ  Setting the version of example-crate to 1.0.0
[12:37:08 PM] [semantic-release] [@semantic-release-cargo/semantic-release-cargo] › ℹ  Upgrading dependency of example-crate-2 to example-crate@1.0.0
```

#### Publish stage
```
[12:37:08 PM] [semantic-release] [@semantic-release-cargo/semantic-release-cargo] › ℹ  Publishing version 1.0.0 of example-crate to crates.io registry
.... build steps
[12:37:08 PM] [semantic-release] [@semantic-release-cargo/semantic-release-cargo] › ℹ  Published example-crate@1.0.0 to crates.io registry
```

and with a private registry

```
[12:37:08 PM] [semantic-release] [@semantic-release-cargo/semantic-release-cargo] › ℹ  Publishing version 1.0.0 of example-crate to your-private-registry registry
.... build steps
[12:37:08 PM] [semantic-release] [@semantic-release-cargo/semantic-release-cargo] › ℹ  Published example-crate@1.0.0 to your-private-registry registry
```

### Debug Logs
Debug logs would then be repurposed to output the entirety of the cargo commands executed

#### Publish stage
```
[12:37:08 PM] [semantic-release] [@semantic-release-cargo/semantic-release-cargo] › ℹ  Publishing version 1.0.0 of example-crate to crates.io registry
[12:37:08 PM] [semantic-release] [@semantic-release-cargo/semantic-release-cargo] › d  cargo publish --manifest-path /build/Cargo.toml --allow-dirty
.... build steps
[12:37:08 PM] [semantic-release] [@semantic-release-cargo/semantic-release-cargo] › ℹ  Published example-crate@1.0.0 to crates.io registry
```

and with a private registry

```
[12:37:08 PM] [semantic-release] [@semantic-release-cargo/semantic-release-cargo] › ℹ  Publishing version 1.0.0 of example-crate to your-private-registry registry
[12:37:08 PM] [semantic-release] [@semantic-release-cargo/semantic-release-cargo] › d  cargo publish --manifest-path /build/Cargo.toml --allow-dirty --registry your-private-registry
.... build steps
[12:37:08 PM] [semantic-release] [@semantic-release-cargo/semantic-release-cargo] › ℹ  Published example-crate@1.0.0 to your-private-registry registry
```

**Comment by ncatelli** [comment]
Additionally it'd be nice to add a mutually-exclusive `--log-level` flag that overrides the `verbosity` flag if set.

As it currently stands
- no `-v` is effectively error level
- `-v` would be info
- etc...

**Comment by jwiesler** [comment]
Are you getting any logs at all @hasezoey? I'm using `cycjimmy/semantic-release-action` and I dont get any output.

On my pipeline, cargo publish is failing with status 101 and at least [this](https://github.com/semantic-release-cargo/semantic-release-cargo/blob/master/src/lib.rs#L722C9-L726C11) log message should be hit. However, there is not a single log message printed from this crate.

I'm pretty stuck with debugging why this could fail since there is no log output.

**Comment by hasezoey** [comment]
> Are you getting any logs at all @hasezoey? I'm using cycjimmy/semantic-release-action and I dont get any output.

no, i dont get any logs from the *plugin* at all.

[example logs](https://github.com/hasezoey/new_string_template/actions/runs/8583201330/job/23522170468) (may not be valid forever)

**Comment by jwiesler** [comment]
Yes that's what I meant, sorry I wasn't very precise.

Then you have the same problem as me. You are not seeing no log messages because there aren't any but because they are (as far as I see it) not forwarded. 
1. Could it be that when loading from semantic-release (loading the js file) the log is not even initialized since the main method is not called? 
2. Could it be that the log messages are not forwarded because we aren't using the logger or stdin/stderr from the context parameter passed in [here](https://github.com/semantic-release-cargo/semantic-release-cargo/blob/master/index.js#L28)?

**Comment by ncatelli** [comment]
I just saw these responses after writing out https://github.com/semantic-release-cargo/semantic-release-cargo/issues/379. Logging with napi-rs, which logs slightly differently through a passed [Writer](https://github.com/semantic-release-cargo/semantic-release-cargo/blob/master/src/lib.rs#L262-L265) for all the stages is inconsistent between the library uses and main. I opened the above issue to start discussing ways to unify that across all the implementations.
